src/execution_plan/execution_plan.c

```diff
@@ -244,101 +245,51 @@ ExecutionPlan* NewExecutionPlan(RedisModuleCtx *ctx,
     if(ast->matchNode) {
         BuildQueryGraph(gc, q, ast->matchNode->_mergedPatterns);
 
-        // For every pattern in match clause.
-        size_t patternCount = Vector_Size(ast->matchNode->patterns);
-        
-        /* Incase we're dealing with multiple patterns
-         * we'll simply join them all together with a join operation. */
-        bool multiPattern = patternCount > 1;
+        int component_count;
+        Node **starting_points = QueryGraph_ConnectedComponents(q, &component_count);
+
+        AE_Unit ***components = AlgebraicExpression_BuildExps(ast, q, starting_points, component_count);
```

Comment:
I'll prefer AlgebraicExpression_BuildExps to receive the query graph, it will internally call QueryGraph_ConnectedComponents

Reply:
:+1: that's a good idea